Federated Catalog sql migrations are missing

## Describe the bug

We're not creating tables needed for the federated catalog.
At this point we're not testing it either I guess.
Is the FC capabilities of the connector used by anybody?

### To Reproduce

Run any test with SQL persistence, some logs will appear, like:
```
[PROVIDER] SEVERE 2025-05-14T09:59:30.211987484 Unexpected error during plan execution
org.eclipse.edc.spi.persistence.EdcPersistenceException: org.postgresql.util.PSQLException: ERROR: relation "edc_target_node_directory" does not exist
  Position: 15
	at org.eclipse.edc.sql.SqlQueryExecutor.query(SqlQueryExecutor.java:97)
	at org.eclipse.edc.catalog.directory.sql.SqlTargetNodeDirectory.lambda$getAll$0(SqlTargetNodeDirectory.java:48)
	at org.eclipse.edc.transaction.local.LocalTransactionContext.execute(LocalTransactionContext.java:74)
	at org.eclipse.edc.catalog.directory.sql.SqlTargetNodeDirectory.getAll(SqlTargetNodeDirectory.java:45)
	at org.eclipse.edc.catalog.cache.ExecutionManager.fetchWorkItems(ExecutionManager.java:169)
	at org.eclipse.edc.catalog.cache.ExecutionManager.doWork(ExecutionManager.java:91)
	at org.eclipse.edc.catalog.cache.ExecutionManager.lambda$executePlan$1(ExecutionManager.java:75)
	at org.eclipse.edc.catalog.crawler.RecurringExecutionPlan.lambda$catchExceptions$0(RecurringExecutionPlan.java:79)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: org.postgresql.util.PSQLException: ERROR: relation "edc_target_node_directory" does not exist
  Position: 15
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2733)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2420)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:372)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:517)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:434)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:194)
	at org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:137)
	at org.eclipse.edc.sql.SqlQueryExecutor.query(SqlQueryExecutor.java:86)
```

### Expected behavior

No exception

## Context Information

current main branch

--- a/edc-controlplane/edc-controlplane-postgresql-hashicorp-vault/build.gradle.kts
+++ b/edc-controlplane/edc-controlplane-postgresql-hashicorp-vault/build.gradle.kts
@@ -26,6 +26,11 @@ plugins {
     alias(libs.plugins.shadow)
 }
 
+configurations.all {
+    // target-node-directory-sql excluded because we provide our own file based target node directory implementation
+    exclude(group = "org.eclipse.edc", module = "target-node-directory-sql")
+}
+
 dependencies {
     runtimeOnly(project(":edc-controlplane:edc-controlplane-base"))
 

